### PR TITLE
rgw: log refactoring for data sync

### PR DESCRIPTION
**Pull request by cbodley**
removes the RGWDataSyncDebugLogger, which duplicates most of the logging from the sync trace framework

for logging outside of sync tracing, this implements DoutPrefixProvider for both RGWBucketSyncStatusManager and RGWDataSyncStatusManager and replaces their ldout()s with ldpp_dout()

### CODE REVIEW

**Comment by cbodley** [comment]
http://pulpito.ceph.com/cbodley-2018-09-05_18:33:58-rgw:multisite-wip-multisite-testing-distro-basic-smithi/

unrelated test failures and valgrind issues

**Comment by vumrao** [comment]
I have created the tracker for this - http://tracker.ceph.com/issues/35830

**Comment by smithfarm** [comment]
@cbodley @vumrao Is this really needed for luminous, and if so, why?

**Comment by cbodley** [comment]
@smithfarm not all of it, no. I think @vumrao is only interested in one piece, and i'll split that into a separate pr

**Comment by cbodley** [comment]
i retargeted that tracker issue at https://github.com/ceph/ceph/pull/23981

File: src/rgw/rgw_data_sync.cc [hunk 1]
```diff
@@ -1479,10 +1405,6 @@ class RGWDataSyncShardCR : public RGWCoroutine {
           yield wait_for_child();
           int ret;
           while (collect(&ret, lease_stack.get())) {
-            if (ret < 0) {
-              ldout(sync_env->cct, 0) << "ERROR: a sync operation returned error" << dendl;
-              /* we have reported this error */
-            }
```
Comment (cbodley) [inline src/rgw/rgw_data_sync.cc]: todo: remove this commit

**Comment by cbodley** [comment]
http://pulpito.ceph.com/cbodley-2018-09-17_14:26:55-rgw:multisite-wip-multisite-testing-distro-basic-smithi/

**Comment by alimaredia** [comment]
@cbodley: The second test you just ran didn't pass `test_bucket_index_log_trim`. Is that related to the PR?

**Comment by cbodley** [comment]
@alimaredia i don't think so, no. i see that failure once in a while, i guess there's a race in the test itself but haven't had time to dig into it

**Pull request merged**